cpp/src/cider/common/Arena.h

```diff
@@ -40,35 +54,137 @@ namespace cider::hashtable {
  */
 class Arena : private boost::noncopyable {
```

Comment:
Will Arena be used under multi thread case?

Reply:
It should be only single thread in our cases.